tests/test_snapshots.py

```
@@ -334,3 +338,186 @@ def test_write_snapshots_handles_errors() -> None:
+        period_start=datetime(2024, 1, 1, tzinfo=UTC),
+        period_end=datetime(2024, 1, 31, tzinfo=UTC),
+    )
+
+    written_paths = [path for _, _, path, _ in github_writer.files_written]
+    assert any("events.json" in p for p in written_paths)
+
+
+def test_raw_events_graceful_when_storage_missing_method() -> None:
+    """events.json is omitted gracefully if storage lacks list_contributions."""
+    class MinimalStorage:
+        def list_recent_notifications(self, limit: int = 1000) -> list[dict]:
```

Ruff (0.15.7)

[warning] Unused method argument: `limit`

(ARG002)